content/articles/object-detection-with-yolov5-and-pytorch/index.md

```diff
@@ -0,0 +1,197 @@
+```bash
+git clone https://github.com/ultralytics/yolov5
+```
+After successfully cloning the repo in our project's environment, we will install all the dependencies using the following command:
+```bash
+cd yolov5
+
+pip install -r requirements.txt
+```
+The command above installs the following packages:
+
+![requirements](/engineering-education/building-an-object-detection-model-with-yolov5-and-pytorch/requirements.png)
+
+Once we're done with installing all the required dependencies, we will move on to training our model.
+
+
+### Model inference using `detect.py`
+
+Since YOLOv5 is a pre-trained model, we will test its performance by running the following command to see how accurately it detects objects through the webcam.
+
+```bash
+python detect.py --source 0
+```
+![object1](/engineering-education/building-an-object-detection-model-with-yolov5-and-pytorch/object1.png)
+
+To test it on other sources, use the following commands:
+
+```bash
+python detect.py --source img.jpg  # for a specific image file
+```
+**Example**
+Running the following command  will detect objects on our images stored in the path `data/images` :
+```bash
+python detect.py --weights yolov5s.pt --img 640 --conf 0.25 --source data/images
+```
+![boda](/engineering-education/building-an-object-detection-model-with-yolov5-and-pytorch/boda1.jpg)
+
+The above image objects will be detected as shown below:
+
+![boda](/engineering-education/building-an-object-detection-model-with-yolov5-and-pytorch/boda.jpg)
+
+>**N/B:** The detected image results are stored in the path `runs/detect/exp`.
+
+For other objects, use the following commands:
+
+```bash
+python detect.py --source vid.mp4  # for a specific video
+python detect.py --source path/  # for a specific directory
+python detect.py --source path/*.jpg  # for a specific glob
+python detect.py --source 'https://youtu.be/Zgi9g1ksQHc'  # for a specific YouTube video
+python detect.py --source 'rtsp://example.com/media.mp4'  # for a specific RTSP, RTMP, or HTTP stream
+```
+
+>**N/B**: Before running the commands above, make sure the objects you want to detect are located at `data/...` e.g. `data/videos`, `data/images` etc.
+
+### Model inference with PyTorch Hub and YOLOv5
+
+To check the inference using PyTorch, we will load the pre-trained YOLOv5s model from PyTorch Hub and then pass an image for inference. 
+
+>N|B: `'YOLOv5s'` is the fastest and lightest YOLOv5 model.
+
+```python
+import torch
+
+# Model
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
+
+# Image
+img = 'https://ultralytics.com/images/zidane.jpg'
+
+# Inference
+results = model(img)
+
+results.pandas().xyxy[0]
+```
+
+To see the results of the code above, feel free to access it on [GitHub.](https://github.com/FREDERICO23/object-detection)
+
+### Model training 
+
+Since the COCO dataset is automatically downloaded from [YOLOv5 latest releases](https://github.com/ultralytics/yolov5/releases), we will proceed to train our model.
```

Comment:
Do you mean to download the entire source code? Or any particular file in the release?